packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -843,6 +868,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       widget.focusNode.addListener(_handleFocusChanged);
       updateKeepAlive();
     }
+    if (widget.readOnly) {
```

Comment:
This is here to close the keyboard if the Editable switched from non-readOnly to readOnly? Can you add a test case for that scenario?

Reply:
What if the opposite happens (going from readOnly to editable)?

Reply:
I will add a test.

Right now the current behavior is when you on focus on the text input in readOnly mode and then switch to non-readonly, the focus is still there attached without the keyboard pop up. The prompt will get to the start of the textfield. if you click anywhere in the text again, the keyboard will pop up. I feel the behavior is ok. I can make the keyboard pop up right after update. I am not sure which one is more nature.

Reply:
added test 'Dynamically switching to read only should close input connection'

Reply:
ios and android does not have build in readonly textfield. It is a web thing, I updated it so the keyboard come up when switches